link/kprobe.go

```diff
@@ -49,6 +49,13 @@ type KprobeOptions struct {
 	//
 	// Needs kernel 5.15+.
 	Cookie uint64
+
+	// Fallback to legacy tracefs kprobe interface
+	// event if kernel has kprobe PMU
+	//
+	// This allows old kernels which has kprobe PMU
+	// but don't support `.` in symbol names to work properly
```

Comment:
are there kernels that supports PMU and don't support `.` in the symbol name and kernels that support PMU and `.` in the symbol name? (I am pretty sure on my kernel (5.16.x) I can attach uprobes PMU with `.` in the symbol)

If yes, what about implementing some feature detection to do this transparently? (kernel supports PMU && kernel doesn't support `.` -> go for tracefs)

Reply:
> are there kernels that supports PMU and don't support . in the symbol name and kernels that support PMU and . in the symbol name?

Centos 8 with kernel v4.18. Did not find the related commit.

Reply:
> If yes, what about implementing some feature detection to do this transparently? (kernel supports PMU && kernel doesn't support . -> go for tracefs)

Don't know how to reliably implement such detection. :(